src/main/java/com/greedobank/cards/model/BankAccount.java

```diff
@@ -0,0 +1,58 @@
+package com.greedobank.cards.model;
+
+import com.fasterxml.jackson.annotation.JsonBackReference;
+import com.greedobank.cards.utils.CurrencyType;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToOne;
+import java.math.BigDecimal;
+import java.time.OffsetDateTime;
+
+@Entity
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class BankAccount {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "bank_account_id", updatable = false, nullable = false)
+    private Integer id;
+
+    @Column(name = "iban")
+    private String iban;
+
+    @Column(name = "balance")
+    private BigDecimal balance;
+
+    @Enumerated(EnumType.STRING)
+    @Column(name = "currency_type")
+    private CurrencyType currency;
+
+    @Column(name = "created_at")
+    private OffsetDateTime created_at;
+
+    @Column(name = "active")
+    private boolean active;
+
+    @OneToOne(cascade = CascadeType.ALL)
+    @JoinColumn(name = "card_id")
+    private Card card;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "customer_id")
+    @JsonBackReference
```

Comment:
what's the purpose of this?

Reply:
I added   @JsonBackReference because I had StackOverFlow error

Reply:
actually, Jackson marshalled tries to traverse through every field class and supposedly faced circular dependencies. That's one of reasons why we don't respond/show up model classes to clients :)